File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
var divideString = function (s, k, fill) {
    let ret = [];
    for (let i = 0; i < s.length; i += k) {
        ret.push(s.slice(i, i + k));   
    }
    if (ret[ret.length - 1].length != k) {
        let lastone = ret.pop();
        for (let i = lastone.length; i < k; i++) {
            lastone=lastone+fill;
        }
        ret.push(lastone);
    }
    return ret;
};

File: 2138-divide-a-string-into-groups-of-size-k/2138-divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        List<String> retu = new ArrayList<>();
        for (int i = 0; i < s.length(); i += k) {
            if (i + k <= s.length()) {
                retu.add(s.substring(i, i + k));
            }
            else {
                retu.add(s.substring(i)); 
            }
        }
        
        if ( retu.get(retu.size() - 1).length() != k) {
            String lastOne = retu.remove(retu.size() - 1);
            for (int i = lastOne.length(); i < k; i++) {
                lastOne = lastOne + fill;
            }
            retu.add(lastOne);
        }
        return retu.toArray(new String[0]);
    }
}
